drud/ddev
# Add disable_settings_management as config option, fixes #2024, fixes #1806

## The Problem/Issue/Bug:

Advanced developers have been asking for an option to turn off ddev's attempts to manage settings files. 

For example, those who are doing a Drupal multisite don't actually want ddev creating a /drush/drush.yml or a sites/default folder. 

## How this PR Solves The Problem:

* Add config.yaml option disable_settings_management
* Add `ddev config --disable-settings-management`
* Stop doing settings files if set

## Manual Testing Instructions:

* Start with a clean project, with no settings files.
* `ddev config --disable-settings-management`
* Verify that no settings files were created
* `ddev start` and verify that no settings files were created
* `ddev config --disable-settings-management=false`
* `ddev start` and verify that settings files are created

## Automated Testing Overview:

TestDdevStartUnmanagedSettings was added to test this behavior

## Related Issue Link(s):

#2024 requests this feature
#1806 was a complaint about ddev's management of drush.yml breaking multisite setup

## Release/Deployment notes:
